hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -124,12 +134,55 @@ private Event getApiEntrySpanForExitSpan(
         .orElse(null);
   }
 
+  /**
+   * An api_node represents an api call in the trace
+   * It consists of api_entry span and multiple api_exit and internal spans
+   * <p>
+   * This method computes the count of exit calls for any given api (identified by api_entry_span)
+   * This count is a composition of 2 things
+   * 1. link between api_exit_span in api_node to api_entry_span (child of api_exit_span) in another api_node
+   * 2. exit calls to backend from api_exit_span in api_node
+   */
+  Map<ByteBuffer, Integer> computeApiExitCallCount(StructuredTrace trace) {
+    // api_node_index -> api exit call count
+    Map<Integer, AtomicInteger> apiNodeExitCallCount = Maps.newHashMap();
+    ApiTraceGraph apiTraceGraph = ViewGeneratorState.getApiTraceGraph(trace);
+
+    // this set contains events which are the source of any api_node_edge
+    Set<ByteBuffer> callerExitEvents = Sets.newHashSet();
+    for (ApiNodeEventEdge apiNodeEventEdge : apiTraceGraph.getApiNodeEventEdgeList()) {
+      // there will be at max 1 call between any two api_node
+      apiNodeExitCallCount.computeIfAbsent(
+          apiNodeEventEdge.getSrcApiNodeIndex(), v -> new AtomicInteger(0)).incrementAndGet();
+      // srcEventIndex won't be null
+      callerExitEvents.add(trace.getEventList().get(apiNodeEventEdge.getSrcEventIndex()).getEventId());
+    }
+
+    // event -> api exit call count for the corresponding api_node
+    Map<ByteBuffer, Integer> eventToApiExitCallCount = Maps.newHashMap();
+
+    for (int index = 0; index < apiTraceGraph.getNodeList().size(); index++) {
+      ApiNode<Event> apiNode = apiTraceGraph.getNodeList().get(index);
+      // {@link ApiTraceGraph#getApiNodeEventEdgeList()} doesn't consists of edges to backend, so compute them explicitly
+      int backendExitCallCountForApiNode = (int) apiNode
+          .getExitApiBoundaryEvents().stream()
+          .filter(exitEvent -> !callerExitEvents.contains(exitEvent.getEventId())).count();
+
+      int totalExitCallCount =
+          apiNodeExitCallCount.getOrDefault(index, new AtomicInteger(0)).get()
+              + backendExitCallCountForApiNode;
+      apiNode.getEvents().forEach(e -> eventToApiExitCallCount.put(e.getEventId(), totalExitCallCount));
+    }
+    return eventToApiExitCallCount;
+  }
+
   private SpanEventView.Builder generateViewBuilder(
       Event event,
       ByteBuffer traceId,
       Map<ByteBuffer, Event> eventMap,
       Map<ByteBuffer, ByteBuffer> childToParentEventIds,
-      Map<ByteBuffer, Event> exitSpanToCalleeApiEntrySpanMap) {
+      Map<ByteBuffer, Event> exitSpanToCalleeApiEntrySpanMap,
```

Comment:
Is there a reason not to pass a higher level construct like the api trace graph to each call so we don't have to change the whole api for each new use case?

Reply:
Yes, because currently there is no such encapsulating Object, creating one just for this seems an overhead.

Reply:
I think, as part of the trace graph work, capturing some of these things would make sense, and then we'd have that encapsulating object - another spot to revisit when refactoring.

Reply:
Exactly